code/modules/mob/living/simple_animal/hostile/megafauna/hierophant.dm

```diff
@@ -124,6 +124,11 @@ Difficulty: Hard
 	adjustHealth(-L.maxHealth*0.5)
 	L.dust()
 
+/mob/living/simple_animal/hostile/megafauna/hierophant/CanAttack(atom/the_target)
+	. = ..()
+	if(istype(the_target, /mob/living/simple_animal/hostile/asteroid/hivelordbrood)) //ignore temporary targets in favor of more permenant targets
```

Comment:
Don't mobs have a typecache blacklist of what they can/can't attack? It's better than an istype.